crawler/regelleistung.py

```diff
@@ -0,0 +1,297 @@
+    earliest_date = get_earliest_date_if_table_exists(engine, table_name)
+    write_past_entries(engine, table_name, url, earliest_date, max_days_past)
+
+
+def write_new_data_from_latest_date_to_today(engine, url, table_name, latest_data_date):
+    log.info(f"Start writing new data to {table_name}")
+    
+    today_date = datetime.today().date()
+
+    if latest_data_date == (today_date - timedelta(days=1)):
+        log.info(f"Table {table_name} has already the newest data.")
+    else:
+        latest_data_date = (latest_data_date + timedelta(days=1))
+        encountered_problem = False
+        while latest_data_date < today_date and not encountered_problem:
+            try:
+                df = get_df_for_date(url, latest_data_date)
+                with engine.begin() as conn:
+                    df.to_sql(table_name, conn, if_exists="append", index=False)
+                latest_data_date += timedelta(days=1)
+            except sqlalchemy.exc.ProgrammingError as e:
+                _, err_obj, _ = sys.exc_info()
+                if ('psycopg2.errors.UndefinedColumn' in str(err_obj)):
+                    log.info(f"handling {repr(e)} by concat")
+                    write_concat_table(engine, table_name, df)
+                    log.info(f"replaced table {table_name}")
+                else:
+                    log.error(f"Encountered error {e}")
+                    encountered_problem = True
+            except Exception as e:
+                encountered_problem = True
+
+        log.info(f"Finished writing new data to {table_name} with newest date being yesterday {(latest_data_date - timedelta(days=1))}")
+
+
+def write_data_in_table(engine, table_name, url, max_days_past=DEFAULT_MAX_DAYS_PAST, write_additional_past_entries_if_any=True):
+    latest_date = get_latest_date_if_table_exists(engine, table_name)
+    if latest_date is not None:
+        write_new_data_from_latest_date_to_today(engine, url, table_name, latest_date)
+        if write_additional_past_entries_if_any:
+            add_additional_past_entries(engine, table_name, url, max_days_past)
+    else:
+        create_table_and_write_past_data(engine, url, table_name, max_days_past)
+
+
+def write_all_tables(engine):
+    write_data_in_table(engine, TABLE_NAME_FCR_DEMANDS, URL_FCR_DEMANDS)
+    write_data_in_table(engine, TABLE_NAME_FCR_RESULTS, URL_FCR_RESULTS)
+    write_data_in_table(engine, TABLE_NAME_FCR_ANONYM_RESULTS, URL_FCR_ANONYM_RESULTS)
+
+    write_data_in_table(engine, TABLE_NAME_AFRR_DEMANDS, URL_AFRR_DEMANDS)
+    write_data_in_table(engine, TABLE_NAME_AFRR_RESULTS, URL_AFRR_RESULTS)
+    write_data_in_table(engine, TABLE_NAME_AFRR_ANONYM_RESULTS, URL_AFRR_ANONYM_RESULTS)
+
+    write_data_in_table(engine, TABLE_NAME_MFRR_DEMANDS, URL_MFRR_DEMANDS)
+    write_data_in_table(engine, TABLE_NAME_MFRR_RESULTS, URL_MFRR_RESULTS)
+    write_data_in_table(engine, TABLE_NAME_MFRR_ANONYM_RESULTS, URL_MFRR_ANONYM_RESULTS)
```

Comment:
could you also add a `f"SELECT public.create_hypertable('{table_name}', 'from_date', if_not_exists => TRUE, migrate_data => TRUE);"` wrapped in an exception handler, just like the other crawlers have?